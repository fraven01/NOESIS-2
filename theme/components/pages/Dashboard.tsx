import * as React from "react";
import { Card } from "../ui/Card";
import { StatusDot } from "../ui/StatusDot";
import { Skeleton } from "../ui/Skeleton";
import { EmptyState } from "../ui/EmptyState";

/**
 * Dashboard overview page with KPI cards, activity list and quick links.
 * @example
 * <Dashboard kpis={[{label:'Users', value:'1k'}]} activities={[]} quickLinks={[]} />
 */
export interface KPI {
  label: string;
  value: string;
}

export interface Activity {
  id: string;
  text: string;
  status: "success" | "warning" | "error" | "neutral";
}

export interface QuickLink {
  label: string;
  href: string;
}

export interface DashboardProps {
  kpis: KPI[];
  activities: Activity[];
  quickLinks: QuickLink[];
  loading?: boolean;
  error?: string;
}

export const Dashboard: React.FC<DashboardProps> = ({
  kpis,
  activities,
  quickLinks,
  loading = false,
  error,
}) => {
  if (error) {
    return <EmptyState title="Error" text={error} />;
  }

  return (
    <div className="flex flex-col gap-8">
      <section aria-label="KPI cards">
        <div className="grid gap-4 sm:grid-cols-2 lg:grid-cols-4">
          {loading
            ? Array.from({ length: 4 }).map((_, i) => (
                <Skeleton key={i} className="h-24" />
              ))
            : kpis.slice(0, 4).map((kpi) => (
                <Card key={kpi.label} className="flex flex-col">
                  <span className="text-sm text-muted">{kpi.label}</span>
                  <span className="text-2xl font-semibold">{kpi.value}</span>
                </Card>
              ))}
        </div>
      </section>
      <section className="grid gap-8 lg:grid-cols-2">
        <Card>
          <h2 className="mb-4 text-lg font-semibold">Recent Activity</h2>
          {loading ? (
            <Skeleton className="h-32" />
          ) : activities.length ? (
            <ul className="space-y-4">
              {activities.map((activity) => (
                <li key={activity.id} className="flex items-center gap-2">
                  <StatusDot status={activity.status} />
                  <span className="text-sm text-fg">{activity.text}</span>
                </li>
              ))}
            </ul>
          ) : (
            <EmptyState title="No activity" text="You're all caught up" />
          )}
        </Card>
        <Card>
          <h2 className="mb-4 text-lg font-semibold">Quick Links</h2>
          {loading ? (
            <Skeleton className="h-32" />
          ) : quickLinks.length ? (
            <ul className="space-y-2">
              {quickLinks.map((link) => (
                <li key={link.href + link.label}>
                  <a
                    href={link.href}
                    className="text-sm text-accent hover:underline focus:underline"
                  >
                    {link.label}
                  </a>
                </li>
              ))}
            </ul>
          ) : (
            <EmptyState title="No links" text="Nothing here yet" />
          )}
        </Card>
      </section>
    </div>
  );
};

