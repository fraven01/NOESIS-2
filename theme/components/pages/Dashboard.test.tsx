import { render } from "@testing-library/react";
import { axe } from "jest-axe";
import { Dashboard, DashboardProps } from "./Dashboard";

describe("Dashboard", () => {
  const props: DashboardProps = {
    kpis: [
      { label: "Users", value: "1.2k" },
      { label: "Sessions", value: "3.4k" },
      { label: "Errors", value: "23" },
      { label: "Uptime", value: "99.9%" },
    ],
    activities: [
      { id: "1", text: "User signed up", status: "success" },
      { id: "2", text: "Server restarted", status: "warning" },
    ],
    quickLinks: [
      { label: "Projects", href: "#" },
      { label: "Documents", href: "#" },
    ],
  };

  it("matches snapshot and is accessible", async () => {
    const { container } = render(<Dashboard {...props} />);
    expect(container).toMatchSnapshot();
    expect(await axe(container)).toHaveNoViolations();
  });
});
