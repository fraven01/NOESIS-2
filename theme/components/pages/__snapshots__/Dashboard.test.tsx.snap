// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dashboard > matches snapshot and is accessible 1`] = `
<div>
  <div
    class="flex flex-col gap-8"
  >
    <section
      aria-label="KPI cards"
    >
      <div
        class="grid gap-4 sm:grid-cols-2 lg:grid-cols-4"
      >
        <div
          class="rounded-2xl border border-muted bg-bg p-6 shadow flex flex-col"
        >
          <span
            class="text-sm text-muted"
          >
            Users
          </span>
          <span
            class="text-2xl font-semibold"
          >
            1.2k
          </span>
        </div>
        <div
          class="rounded-2xl border border-muted bg-bg p-6 shadow flex flex-col"
        >
          <span
            class="text-sm text-muted"
          >
            Sessions
          </span>
          <span
            class="text-2xl font-semibold"
          >
            3.4k
          </span>
        </div>
        <div
          class="rounded-2xl border border-muted bg-bg p-6 shadow flex flex-col"
        >
          <span
            class="text-sm text-muted"
          >
            Errors
          </span>
          <span
            class="text-2xl font-semibold"
          >
            23
          </span>
        </div>
        <div
          class="rounded-2xl border border-muted bg-bg p-6 shadow flex flex-col"
        >
          <span
            class="text-sm text-muted"
          >
            Uptime
          </span>
          <span
            class="text-2xl font-semibold"
          >
            99.9%
          </span>
        </div>
      </div>
    </section>
    <section
      class="grid gap-8 lg:grid-cols-2"
    >
      <div
        class="rounded-2xl border border-muted bg-bg p-6 shadow"
      >
        <h2
          class="mb-4 text-lg font-semibold"
        >
          Recent Activity
        </h2>
        <ul
          class="space-y-4"
        >
          <li
            class="flex items-center gap-2"
          >
            <span
              aria-hidden="true"
              class="inline-block h-2 w-2 rounded-full bg-accent"
            />
            <span
              class="text-sm text-fg"
            >
              User signed up
            </span>
          </li>
          <li
            class="flex items-center gap-2"
          >
            <span
              aria-hidden="true"
              class="inline-block h-2 w-2 rounded-full bg-muted"
            />
            <span
              class="text-sm text-fg"
            >
              Server restarted
            </span>
          </li>
        </ul>
      </div>
      <div
        class="rounded-2xl border border-muted bg-bg p-6 shadow"
      >
        <h2
          class="mb-4 text-lg font-semibold"
        >
          Quick Links
        </h2>
        <ul
          class="space-y-2"
        >
          <li>
            <a
              class="text-sm text-accent hover:underline focus:underline"
              href="#"
            >
              Projects
            </a>
          </li>
          <li>
            <a
              class="text-sm text-accent hover:underline focus:underline"
              href="#"
            >
              Documents
            </a>
          </li>
        </ul>
      </div>
    </section>
  </div>
</div>
`;
